ocp_utilities/data_collector.py

```diff
@@ -0,0 +1,165 @@
+import logging
+import os
+
+import infra
+from ocp_resources.datavolume import DataVolume
+from ocp_resources.namespace import Namespace
+from ocp_resources.pod import Pod
+from ocp_resources.project import ProjectRequest
+from ocp_resources.service import Service
+from ocp_resources.virtual_machine import VirtualMachine
+from pytest_testconfig import config as py_config
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def collect_vmi_data(vmi, directory=None):
+    write_to_file(
+        file_name=f"vmi-{vmi.name}.yaml",
+        content=vmi.instance.to_str(),
```

Comment:
Say a vm failed to come up, and we are collecting data, there won't be any vmi related info in that case, should we handle that exception, so that data collection does not exit prematurely? I am assuming worst case scenario. Should we also consider collecting node information in such case?